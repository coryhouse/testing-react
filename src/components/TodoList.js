import React from 'react';
import Header from './Header';
import Footer from './Footer';
import AddTodo from './AddTodo';
import TodoLists from './TodoLists';

export default class TodoList extends React.Component {
  state = {
    todos: []
  };
  componentDidMount() {
    const todos = localStorage.getItem('todos');
    if (todos) {
      const parsedJSON = JSON.parse(todos);
      this.setState({ todos: parsedJSON });
    }
  }
  componentDidUpdate(prevProps, prevState) {
    const { todos } = this.state;
    if (prevState.todos.length !== this.state.todos.length) {
      const json = JSON.stringify(todos);
      localStorage.setItem('todos', json);
    }
  }
  handleAddTodo = event => {
    event.preventDefault();
    const todo = event.target.todo.value.trim();
    if (todo) {
      this.setState(prevState => ({
        todos: [...prevState.todos, todo]
      }));
      event.target.todo.value = '';
    }
  };
  render() {
    return (
      <div className="container">
        <Header />
        <AddTodo handleAddTodo={this.handleAddTodo} />
        <TodoLists todos={this.state.todos} />
        <Footer />
      </div>
    );
  }
}
