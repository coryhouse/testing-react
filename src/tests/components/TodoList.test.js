import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../../components/TodoList';

let wrapper;

beforeAll(() => {
  wrapper = shallow(<TodoList />);
});

test('should test TodoList component', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call handleAddTodo method', () => {
  const instance = wrapper.instance();
  const value = 'Publish Article';
  instance.handleAddTodo({
    preventDefault: () => {},
    target: {
      todo: {
        value
      }
    }
  });
});
