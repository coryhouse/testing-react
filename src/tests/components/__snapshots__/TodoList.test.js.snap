// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test TodoList component 1`] = `
<div
  className="container"
>
  <Header />
  <AddTodo
    handleAddTodo={[Function]}
  />
  <TodoLists
    todos={Array []}
  />
  <Footer />
</div>
`;
